### DISCUSSION TITLE
connection timeout

### DISCUSSION THREAD

**Discussion by Vidhya-RajaMani** [jtv/libpqxx#815] [Q&A]
Is there any ways to set timeout while connecting the database?

**Comment by jtv** [comment]
I think you'd have to arrange that at the OS level.  It's a question of how long the networking code is willing to wait for a connection to be established.

**Comment by Vidhya-RajaMani** [comment]
Thanks for the quick response.

In Postgres 14 support connection timeout. If connection is ideal for certain period(given timeout). The connection will be destroyed once timeout is reached.

If any possible to set the timeout during connection in connection string or any other way to set timeout.

`std::shared_ptr<pqxx::connection> connection = std::make_shared<pqxx::connection>("dbname = xxx user = yyy password = 123 hostaddr = 127.0.0.1 port = 8080 connection_timeout = 120");
`

I have tried the above sample. But it doesn't work as expected. 

How to achieve timeout, is there any other ways to set timeout? Please guide me.

Thanks.

**Reply by jtv** [reply]
IIRC there's an interplay between OS-level timeouts; failures that are immediate versus failures that manifest as packets simply never arriving; and retries that libpq does.  So it can take a lot longer than you'd expect, depending on circumstances.

There is very little that libpqxx changes about these things, and that's by design.  Don't want to complicate it further.

**Reply by tt4g** [reply]
`connection_timeout` is wrong. `connect_timeout` is correct.
See: https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-CONNECT-TIMEOUT
